server: TestStartupInjectedFailureSingleNode failed [session based leasing issues]

server.TestStartupInjectedFailureSingleNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/55fd254f-eb71-4570-b9ce-059f0736f1a4?testReportRun=10&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [03525787deade5395bac0a8b5735fa7e715e22ef](https://github.com/cockroachdb/cockroach/commits/03525787deade5395bac0a8b5735fa7e715e22ef):


```
        	            	  |   | 	src/runtime/asm_amd64.s:1650
        	            	  | Wraps: (2) list-tenants
        	            	  | Wraps: (3) unexpected value: raw_bytes:"\372N\360\267\n3\002" timestamp:<wall_time:1706680561085812090 > 
        	            	  | Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	            	Wraps: (4) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1074
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).getExpectedRunningTenants
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:206
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startInitialSecondaryTenantServers.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:99
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startInitialSecondaryTenantServers
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:95
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:43
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2187
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (5) list-tenants
        	            	Wraps: (6) unexpected value: raw_bytes:"\372N\360\267\n3\002" timestamp:<wall_time:1706680561085812090 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *secondary.withSecondaryError (4) *withstack.withStack (5) *errutil.withPrefix (6) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestStartupInjectedFailureSingleNode1240926317
--- FAIL: TestStartupInjectedFailureSingleNode (3.54s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35766

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13765483?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13765483?buildTab=artifacts#/) on master @ [719e3abd2172325b89f9d5b82bf34d19c64d9d92](https://github.com/cockroachdb/cockroach/commits/719e3abd2172325b89f9d5b82bf34d19c64d9d92):


```
        	            	  |   | 	src/runtime/asm_amd64.s:1650
        	            	  | Wraps: (2) list-tenants
        	            	  | Wraps: (3) unexpected value: raw_bytes:"\213U4)\n3\002" timestamp:<wall_time:1706703417625920013 > 
        	            	  | Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	            	Wraps: (4) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1074
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).getExpectedRunningTenants
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:206
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startInitialSecondaryTenantServers.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:99
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startInitialSecondaryTenantServers
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:95
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:43
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2187
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (5) list-tenants
        	            	Wraps: (6) unexpected value: raw_bytes:"\213U4)\n3\002" timestamp:<wall_time:1706703417625920013 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *secondary.withSecondaryError (4) *withstack.withStack (5) *errutil.withPrefix (6) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestStartupInjectedFailureSingleNode2676452708
--- FAIL: TestStartupInjectedFailureSingleNode (34.68s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13768025?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13768025?buildTab=artifacts#/) on master @ [07adffbb0c8efef899f39983d66fb87b962e93b0](https://github.com/cockroachdb/cockroach/commits/07adffbb0c8efef899f39983d66fb87b962e93b0):


```
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1074
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1633
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/aarch64-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1197
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"\245\016\350$\n3\002" timestamp:<wall_time:1706713865247541535 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
--- FAIL: TestStartupInjectedFailureSingleNode (25.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13771883?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13771883?buildTab=artifacts#/) on master @ [c46b7bb5966c30d24abb34873373ed25dcf6648e](https://github.com/cockroachdb/cockroach/commits/c46b7bb5966c30d24abb34873373ed25dcf6648e):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1633
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/aarch64-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1197
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"\226\027\373\252\n3\002" timestamp:<wall_time:1706729481264575050 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f537c08f6697b7d24ac4f4be9defa70f/logTestStartupInjectedFailureSingleNode3058981422
--- FAIL: TestStartupInjectedFailureSingleNode (2.23s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/19943fca-2ba0-4c23-a6b6-53091a43f0a4?testReportRun=3&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [cc4fdffa8532d16544c48ef036689763f737dc6b](https://github.com/cockroachdb/cockroach/commits/cc4fdffa8532d16544c48ef036689763f737dc6b):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1633
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"\247zdc\n3\002" timestamp:<wall_time:1706764852474099312 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestStartupInjectedFailureSingleNode2139810033
--- FAIL: TestStartupInjectedFailureSingleNode (1.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13781289?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13781289?buildTab=artifacts#/) on master @ [cc4fdffa8532d16544c48ef036689763f737dc6b](https://github.com/cockroachdb/cockroach/commits/cc4fdffa8532d16544c48ef036689763f737dc6b):


```
        	            	  |   | 	src/runtime/asm_amd64.s:1650
        	            	  | Wraps: (2) list-tenants
        	            	  | Wraps: (3) unexpected value: raw_bytes:"\312\207\345\005\n3\002" timestamp:<wall_time:1706799449780626334 > 
        	            	  | Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	            	Wraps: (4) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1074
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).getExpectedRunningTenants
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:206
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startInitialSecondaryTenantServers.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:99
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startInitialSecondaryTenantServers
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:95
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:43
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2187
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (5) list-tenants
        	            	Wraps: (6) unexpected value: raw_bytes:"\312\207\345\005\n3\002" timestamp:<wall_time:1706799449780626334 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *secondary.withSecondaryError (4) *withstack.withStack (5) *errutil.withPrefix (6) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestStartupInjectedFailureSingleNode4163893663
--- FAIL: TestStartupInjectedFailureSingleNode (7.92s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13781290?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13781290?buildTab=artifacts#/) on master @ [cc4fdffa8532d16544c48ef036689763f737dc6b](https://github.com/cockroachdb/cockroach/commits/cc4fdffa8532d16544c48ef036689763f737dc6b):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1633
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"M\304L\264\n3\002" timestamp:<wall_time:1706801337407398691 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:541: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestStartupInjectedFailureSingleNode1664611304
--- FAIL: TestStartupInjectedFailureSingleNode (12.12s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://tanzanite.cluster.engflow.com/invocations/default/e28999f8-4d51-4e92-b3d9-a9e402595828?testReportRun=4&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [fce4d4723519bc4ca6e9ef5da0ae19960c84752c](https://github.com/cockroachdb/cockroach/commits/fce4d4723519bc4ca6e9ef5da0ae19960c84752c):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1632
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"]t\034\215\n3\002" timestamp:<wall_time:1706851281523973892 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestStartupInjectedFailureSingleNode1393796343
--- FAIL: TestStartupInjectedFailureSingleNode (2.08s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13798841?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13798841?buildTab=artifacts#/) on master @ [0778173b902ce6d418b843389d20509b934f451d](https://github.com/cockroachdb/cockroach/commits/0778173b902ce6d418b843389d20509b934f451d):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1632
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"=5\025\211\n3\002" timestamp:<wall_time:1706858159231607040 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/63dd67dfe92cccaca0d733d912c3499b/logTestStartupInjectedFailureSingleNode4063179124
--- FAIL: TestStartupInjectedFailureSingleNode (2.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestStartupInjectedFailureSingleNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13802218?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13802218?buildTab=artifacts#/) on master @ [0778173b902ce6d418b843389d20509b934f451d](https://github.com/cockroachdb/cockroach/commits/0778173b902ce6d418b843389d20509b934f451d):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:508
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:567
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:681
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:733
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:719
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:703
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable.CheckIfMigrationCompleted
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/migrationstable/migrations_table.go:118
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/util/startup.RunIdempotentWithRetryEx[...]
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/startup/retry.go:142
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:216
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1632
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*topLevelServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server.go:2053
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:807
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*testServer).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/testserver.go:883
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/serverutils.(*wrap).Start
        	            	  | 	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/testutils/serverutils/ts_control_forwarder_generated.go:15
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1766
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServerWithInspect
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1776
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RestartServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:1690
        	            	  | github.com/cockroachdb/cockroach/pkg/server_test.TestStartupInjectedFailureSingleNode
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server_test/pkg/server/server_startup_test.go:90
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) migration-job-find-already-completed
        	            	Wraps: (3) unexpected value: raw_bytes:"\252\021\256\236\n3\002" timestamp:<wall_time:1706881284260054049 > 
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *kvpb.ConditionFailedError
        	Test:       	TestStartupInjectedFailureSingleNode
        	Messages:   	failed to restart server
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestStartupInjectedFailureSingleNode2943690580
--- FAIL: TestStartupInjectedFailureSingleNode (7.34s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStartupInjectedFailureSingleNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/lease/BUILD.bazel
+++ b/pkg/sql/catalog/lease/BUILD.bazel
@@ -57,6 +57,7 @@ go_library(
         "//pkg/util/metric",
         "//pkg/util/quotapool",
         "//pkg/util/retry",
+        "//pkg/util/startup",
         "//pkg/util/stop",
         "//pkg/util/syncutil",
         "//pkg/util/syncutil/singleflight",

--- a/pkg/sql/catalog/lease/storage.go
+++ b/pkg/sql/catalog/lease/storage.go
@@ -19,7 +19,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv"
-	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/multitenant"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings"
@@ -38,6 +37,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/cockroach/pkg/util/startup"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight"
 	"github.com/cockroachdb/errors"
@@ -138,7 +138,6 @@ func (s storage) acquire(
 	prefix = s.getRegionPrefix()
 	var sessionID []byte
 	acquireInTxn := func(ctx context.Context, txn *kv.Txn) (err error) {
-
 		// Run the descriptor read as high-priority, thereby pushing any intents out
 		// of its way. We don't want schema changes to prevent lease acquisitions;
 		// we'd rather force them to refresh. Also this prevents deadlocks in cases
@@ -179,10 +178,14 @@ func (s storage) acquire(
 			// a monotonically increasing expiration.
 			expiration = minExpiration.Add(int64(time.Millisecond), 0)
 		}
-		desc, err = s.mustGetDescriptorByID(ctx, txn, id)
+		// Read into a temporary variable in case our read runs into
+		// any retryable error. If we run into an error then the delete
+		// above may need to be executed again.
+		latestDesc, err := s.mustGetDescriptorByID(ctx, txn, id)
 		if err != nil {
 			return err
 		}
+		desc = latestDesc
 		if err := catalog.FilterAddingDescriptor(desc); err != nil {
 			return err
 		}
@@ -228,10 +231,9 @@ func (s storage) acquire(
 	// are propagated up to the caller.
 	for r := retry.StartWithCtx(ctx, retry.Options{}); r.Next(); {
 		err := s.db.KV().Txn(ctx, acquireInTxn)
-		var pErr *kvpb.AmbiguousResultError
 		switch {
-		case errors.As(err, &pErr):
-			log.Infof(ctx, "ambiguous error occurred during lease acquisition for %v, retrying: %v", id, err)
+		case startup.IsRetryableReplicaError(err):
+			log.Infof(ctx, "retryable replica error occurred during lease acquisition for %v, retrying: %v", id, err)
 			continue
 		case pgerror.GetPGCode(err) == pgcode.UniqueViolation:
 			log.Infof(ctx, "uniqueness violation occurred due to concurrent lease"+

